src/components/charts/helper.js

```diff
@@ -47,29 +47,47 @@ const getCategoryColorFrom2Colors = index => colorsForCharts.categorical2Colors[
 
 const getDataKey = showPercentage => (showPercentage ? 'percentage' : 'value');
 
-const transformArrangerDataToChart = (field) => {
+const transformArrangerDataToChart = (field, sqonValues) => {
   const chartData = [];
-  field.buckets.map(bucket =>
-    chartData.push({
-      name: bucket.key,
-      value: bucket.doc_count,
-    }),
-  );
+  field.buckets
+    .filter(bucket => (sqonValues === undefined || sqonValues.includes(bucket.key)))
+    .forEach(bucket =>
+      chartData.push({
+        name: bucket.key,
+        value: bucket.doc_count,
+      }),
+    );
   return chartData;
 };
 
-const transformArrangerDataToSummary = (field, chartType, title) => ({
+const transformArrangerDataToSummary = (field, chartType, title, sqonValues) => ({
   type: chartType,
   title,
-  data: transformArrangerDataToChart(field),
+  data: transformArrangerDataToChart(field, sqonValues),
 });
 
-const transformDataToCount = (field, label) => ({
+const transformDataToCount = (field, label, sqonCount) => ({
   label,
-  value: field.buckets.length,
+  value: Math.min(field.buckets.length, sqonCount),
 });
 
-const getCharts = (data, arrangerConfig) => {
+const getSQONValues = (sqon, field) => {
+  if (!sqon || !sqon.content) return undefined;
+  const sqonItems = sqon.content.filter(item => item.content.field === field);
+  if (!sqonItems || sqonItems.length !== 1) return undefined;
+  const sqonValues = sqonItems[0].content.value;
```

Comment:
It seems arranger use `sqon` for filtering queries, `sqon.content` stores all selected fields, and since this function only need one specific field, the `sqonItems` should be always in length of 1 or zero: 1 means there are some values selected on this field, and zero means no values selected on this field (which means every values should be counted for this field in the visualizations)